File: go/gigasecond/gigasecond.go
// Determine the date when a gigasecond has passed from a given date.
package gigasecond

// import path for the time package from the standard library
import "time"

// For a given time return that time with a billion seconds added to it.
// @param t: The time to add to.
// @returns: Time when a gigasecond interval has passed.
func AddGigasecond(t time.Time) time.Time {
	const seconds = 1_000_000_000
	var duration time.Duration = time.Second * seconds
	return t.Add(duration)
}

File: php/gigasecond/Gigasecond.php
<?php
// Gigasecond exercise.

declare(strict_types=1);

const GIGA = 1000000000;

// Calcluate a gigasecond after the given date.
// @param $date: The date to add a gigasecond to.
// @returns: DateTime value a gigasecond after the given date.
function from(DateTimeImmutable $date): DateTimeImmutable
{
    return $date->add(new DateInterval("PT" . strval(GIGA) . "S"));
}
